torchqdynamics/utils/utils.py

```diff
@@ -22,95 +22,106 @@
 ]
 
 
-def is_ket(x: Tensor) -> Tensor:
-    """Check if a tensor is in state vector format."""
+def is_ket(x: Tensor) -> bool:
+    """Check if a tensor is in state vector format.
+
+    Args:
+        x ( Tensor (..., d, n) ): State vector, linear map or density matrix.
+
+    Returns:
+        `True` if the last dimension `n` is `1`, `False` otherwise.
+    """
     return x.size(-1) == 1
 
 
 def ket_to_bra(x: Tensor) -> Tensor:
-    """Linear map (bra) representation of a state vector (ket).
+    r"""Linear map $\bra\psi$ representation of a state vector $\ket\psi$.
 
     Args:
-        x: Tensor of dimension `(..., d, 1)`.
+        x ( Tensor (..., d, 1) ): State vector.
```

Comment:
```suggestion
        x ( Tensor (..., n, 1) ): State vector.
```

Reply:
Do you propose to use `n` instead of `d` for all the Hilbert space dimensions?

Reply:
Yes, I would argue it's better. It's closer to numpy and pytorch notations which uses "n" or "N". See e.g. https://numpy.org/doc/stable/reference/generated/numpy.ndarray.put.html#numpy.ndarray.put and https://pytorch.org/docs/stable/generated/torch.chain_matmul.html.